### ISSUE TITLE
@unsafe blocks don't work inside if-statements

### ISSUE THREAD

**Issue by kiiriis** [shuaimu/rusty-cpp#12]
# Bug Report: @unsafe blocks don't work inside if-statements

## Summary

`@unsafe` blocks are not detected when they are nested inside `if` statements. The same `@unsafe` block works correctly when used standalone (not inside any control structure).

## Minimal Reproduction

```cpp
// File: rusty-cpp-unsafe-block-test.cpp
#include <thread>

class Test {
    std::thread t_;

public:
    // @safe - WORKS: standalone @unsafe block with method call
    void test_standalone_works() {
        // @unsafe
        {
            t_.detach();  // No violation - @unsafe block works
        }
    }

    // @safe - FAILS: @unsafe block INSIDE if-statement with method call
    void test_inside_if_fails() {
        if (t_.joinable()) {
            // @unsafe
            {
                t_.detach();  // VIOLATION despite @unsafe block!
            }
        }
    }
};
```

## How to Reproduce

```bash
./third-party/rusty-cpp/target/release/rusty-cpp-checker ./rusty-cpp-unsafe-block-test.cpp
```

## Expected Behavior

Both functions should pass without violations because both `t_.detach()` calls are inside `@unsafe` blocks.

## Actual Behavior

```
Rusty C++ Checker
Analyzing: rusty-cpp-unsafe-block-test.cpp
✗ Found 1 violation(s) in rusty-cpp-unsafe-block-test.cpp:
In function 'Test::test_inside_if_fails': Calling undeclared function 'std::thread::detach' at line 23 - must be explicitly marked @safe or @unsafe
```

- `test_standalone_works()` - **PASSES** (no violation)
- `test_inside_if_fails()` - **FAILS** (violation reported despite @unsafe block)

## Root Cause Analysis

**The bug is in `src/analysis/unsafe_propagation.rs` lines 180-202.**

When checking `Statement::If` branches, the code passes `in_unsafe_scope` to recursive calls but **does NOT process `EnterUnsafe`/`ExitUnsafe` markers** inside those branches:

```rust
Statement::If { condition, then_branch, else_branch, location } => {
    // ...
    // Recursively check branches
    for branch_stmt in then_branch {
        if let Some(error) = check_statement_for_unsafe_calls_with_external(
            branch_stmt, safety_context, known_safe_functions,
            external_annotations, template_params, in_unsafe_scope  // <-- BUG: in_unsafe_scope never updated!
        ) {
            return Some(error);
        }
    }
    // ...
}
```

The `then_branch` Vec contains `EnterUnsafe`/`ExitUnsafe` statements (inserted by the parser), but this loop doesn't track them. It just passes the same `in_unsafe_scope` value to every statement.

**Compare to the top-level function check (lines 31-55)** which correctly tracks unsafe depth:

```rust
for stmt in &function.body {
    match stmt {
        Statement::EnterUnsafe => {
            unsafe_depth += 1;  // <-- Properly tracked at top level
            continue;
        }
        Statement::ExitUnsafe => {
            if unsafe_depth > 0 {
                unsafe_depth -= 1;
            }
            continue;
        }
        _ => {}
    }
    let in_unsafe_scope = unsafe_depth > 0;
    // ... check statement with updated in_unsafe_scope
}
```
## Test File

- `rusty-cpp-unsafe-block-test.cpp` - Minimal reproduction case
